core/internal/leet/model.go

```diff
@@ -208,24 +414,135 @@ func (m *Model) renderStatusBar() string {
+	if filterInfo == "" {
+		filterInfo = "no matches"
+	}
+	return fmt.Sprintf(" Overview filter: %s_ [%s] (@e/@c/@s for sections • Enter to apply)",
+		m.leftSidebar.FilterQuery(), filterInfo)
+}
+
+// buildMetricsFilterStatus builds status for metrics filter mode.
+func (m *Model) buildMetricsFilterStatus() string {
+	matchCount := m.metricsGrid.effectiveChartCountNoLock()
+	m.metricsGrid.mu.RLock()
+	totalCount := len(m.metricsGrid.all)
+	m.metricsGrid.mu.RUnlock()
+	return fmt.Sprintf(" Filter: %s_ [%d/%d matches] (Enter to apply)",
+		m.metricsGrid.filter.draft, matchCount, totalCount)
+}
+
+// buildGridConfigStatus builds status for grid configuration mode.
+func (m *Model) buildGridConfigStatus() string {
+	switch m.pendingGridConfig {
+	case gridConfigMetricsCols:
+		return " Press 1-9 to set metrics grid columns (ESC to cancel)"
+	case gridConfigMetricsRows:
+		return " Press 1-9 to set metrics grid rows (ESC to cancel)"
+	case gridConfigSystemCols:
+		return " Press 1-9 to set system grid columns (ESC to cancel)"
+	case gridConfigSystemRows:
+		return " Press 1-9 to set system grid rows (ESC to cancel)"
+	default:
+		return ""
+	}
+}
+
+// buildLoadingStatus builds status for loading mode.
+func (m *Model) buildLoadingStatus() string {
+	m.metricsGrid.mu.RLock()
+	chartCount := len(m.metricsGrid.all)
+	m.metricsGrid.mu.RUnlock()
+
+	if m.recordsLoaded > 0 {
+		return fmt.Sprintf(" Loading data... [%d records, %d metrics]",
+			m.recordsLoaded, chartCount)
+	}
+	return " Loading data..."
+}
+
+// buildActiveStatus builds status for active (non-loading, non-filter) mode.
+func (m *Model) buildActiveStatus() string {
+	var parts []string
+
+	// Add filter info if active.
+	if m.metricsGrid.filter.applied != "" {
+		m.metricsGrid.mu.RLock()
+		filteredCount := len(m.metricsGrid.filtered)
+		totalCount := len(m.metricsGrid.all)
+		m.metricsGrid.mu.RUnlock()
+		parts = append(parts, fmt.Sprintf("Filter: \"%s\" [%d/%d] (/ to change, Ctrl+L to clear)",
+			m.metricsGrid.filter.applied, filteredCount, totalCount))
+	}
+
+	// Add overview filter info if active.
+	if m.leftSidebar.IsFiltering() {
+		parts = append(parts, fmt.Sprintf("Overview: \"%s\" [%s] (o to change, Ctrl+K to clear)",
+			m.leftSidebar.FilterQuery(),
+			m.leftSidebar.FilterInfo(),
+		))
+	}
+
+	// Add selected overview item if sidebar is visible.
+	if m.leftSidebar.IsVisible() {
+		key, value := m.leftSidebar.SelectedItem()
+		if key != "" {
+			parts = append(parts, fmt.Sprintf("%s: %s", key, value))
 		}
 	}
-	return false, nil
+
+	// Add focused metric name if a chart is focused.
+	focusedTitle := m.FocusedTitle()
+	if focusedTitle != "" {
+		parts = append(parts, focusedTitle)
+	}
+
+	if len(parts) == 0 {
+		return " "
+	}
+
+	return " " + strings.Join(parts, " • ")
+}
+
+// buildHelpText builds the help text for the status bar.
+func (m *Model) buildHelpText() string {
+	if !m.metricsGrid.filter.inputActive && !m.leftSidebar.filter.inputActive {
+		return "h: help "
```

Comment:
`buildStatusText` always prepends a space at the start of the string, but this adds a space at the end; is that intentional?

Reply:
Embarrassingly, this was some janky padding on the sides of the status bar :) Implemented it properly now without any trailing or leading spaces.